docs/src/hugo/content/documentation/configuration.md

```diff
@@ -0,0 +1,691 @@
+
+All configuration in Nelson is a sub-config of the `nelson` scope / namespace. This can either be specified using "block" sytanx, or "inline" syntax. Consider the following example:
+
+```
+# nested block syntax
+nelson {
+  timeout = 4 seconds
+}
+
+# which can also be written using inline syntax:
+nelson.timeout = 4 seconds
+```
+
+These are 100% functionally equivalent, and there is no prescription on what style you use where in your deployment of Nelson. This guide uses inline syntax for referencing specific fields, simply for compactness and readability in this guide. It is however recomended that the block syntax be used in production deployments, as the Nelson team believe it is more readable for most users.
+
+----
+
+## Core
+
+Some configuration in Nelson is "top-level" in the `nelson` scope. These options are typically core to how Nelson operates, and Nelson administrators should carefully review these options to ensure that they are set to values that make sense for your deployment.
+
+* [nelson.default-namespace](#nelson-default-namespace)
+* [nelson.discovery-delay](#nelson-discovery-delay)
+* [nelson.manifest-filename](#nelson-manifest-filename)
+* [nelson.proxy-port-whitelist](#nelson-proxy-port-whitelist)
+* [nelson.readiness-delay](#nelson-readiness-delay)
+* [nelson.timeout](#nelson-timeout)
+
+#### nelson.default-namespace 
+
+upon receiving a github release event, where should Nelson assume the
+application should get deployed too. This can either be a root namespace,
+or a subordinate namespace, e.g. `stage/unstable`... its arbitrary, but the
+namespace must exist (otherwise Nelson will attempt to create it on boot)
+
+```
+nelson.default-namespace = "dev"
+```
+
+#### nelson.discovery-delay
+
+The frequency on which should nelson write out the discovery / routing protocol information to your configured routing sub-system. This should typically not be set too high as it will affect the frequency and choppiness of updates in the runtime. For example, if you set the value to 10 minutes and wanted to migrate traffic in 30 minutes, you'd end up with 3 large "steps" in that traffic-shifting curve (assuming a linear specification). 
+
+```
+nelson.discovery-delay = 2 minutes
+```
+
+#### nelson.manifest-filename
+
+When Nelson fetches the repository manifest, this field controls what filename Nelson should be looking for. By default, the expectation is that this config is a path from the root of the repository, but can be specified with relative paths (from the root of any given repository).
+
+```
+nelson.manifest-filename = ".nelson.yml"
+```
+
+#### nelson.proxy-port-whitelist
+
+When using Nelson [load balancer](/getting-started/routing.html#load-balancers) feature, to avoid exposing random ports on the internet, administrators can keep a whitelist of the ports they are permitting to be exposed. This helps mitigate issues with security surface areas, whilst not restricting users to some hardcoded defaults.  
```

Comment:
```suggestion
When using Nelson's [load balancer](/getting-started/routing.html#load-balancers) feature, to avoid exposing random ports on the internet, administrators can keep a whitelist of the ports they are permitting to be exposed. This helps mitigate issues with security surface areas, whilst not restricting users to some hardcoded defaults.  
```